lib/galaxy/datatypes/microarrays.py

```diff
@@ -0,0 +1,152 @@
+import logging
+import subprocess
+
+from galaxy.datatypes import data
+from galaxy.datatypes.metadata import MetadataElement
+
+log = logging.getLogger(__name__)
+
+
+def get_n_first_lines(filename, n=3):
+    """
+        Rendering n first lines in order to populate peek.
+    """
+    try:
+        cmd = ["head", '-' + str(n)]
+        cmd.extend([filename])
+        out = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        return out.communicate()[0]
+    except Exception:
+        pass
+    return 0
+
+
+def count_special_lines(word, filename, invert=False):
+    """
+        searching for special 'words' using the grep tool
+        grep is used to speed up the searching and counting
+        The number of hits is returned.
+    """
+    try:
+        cmd = ["grep", "-c"]
+        if invert:
+            cmd.append('-v')
+        cmd.extend([word, filename])
+        out = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        return int(out.communicate()[0].split()[0])
+    except Exception:
+        pass
+    return 0
+
+
+def count_lines(filename, non_empty=False):
+    """
+        counting the number of lines from the 'filename' file
+    """
+    try:
+        out = subprocess.Popen(['wc', '-l', filename], stdout=subprocess.PIPE)
+        return int(out.communicate()[0].split()[0])
+    except Exception:
+        pass
+    return 0
+
+
+class GenericMicroarrayFile(data.Text):
+    """
+    Abstract class for most of the microarray files.
+    """
+    MetadataElement(name="number_of_comments", default=1, desc="Number of comments", readonly=True, visible=True,
+                    optional=True, no_value=0)
+    MetadataElement(name="number_of_blocks", default=1, desc="Number of blocks", readonly=True, visible=True,
+                    optional=True, no_value=0)
+    MetadataElement(name="number_of_records", default=1, desc="Number of records", readonly=True, visible=True,
+                    optional=True, no_value=0)
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            if dataset.metadata.number_of_blocks == 1:
+                dataset.blurb = "1 block %s recodrs" % dataset.metadata.number_of_records
+                dataset.peek = get_n_first_lines(dataset.file_name, n=3)
```

Comment:
This line is repeated in the else branch, so it can be removed if you apply my suggestion below.